subsys/bluetooth/host/adv.c

```diff
@@ -997,6 +997,16 @@ int bt_le_adv_start_ext(struct bt_le_ext_adv *adv,
 
 static void adv_timeout(struct k_work *work);
 
+void bt_le_lim_adv_cancel_timeout(struct bt_le_ext_adv *adv)
+{
+	int rc;
+
+	(void)k_work_cancel_delayable(&adv->timeout_work);
+	rc = k_work_delayable_busy_get(&adv->timeout_work) != 0;
+	/* At this point timeout must be canceled*/
+	__ASSERT(rc == 0, "Failed to stop limited advertising timeout");
```

Comment:
IMO, it'd be better to use a flag for this than asserting. I'm not confident that this couldn't fail.

The most common pattern for handling cancel failures is to set a flag right after the timer is started and checking it in the beginning of the timeout handler. In this cancel function, you can clear the flag and cancel the delayable. If the cancel call fails, the flag will prevent misfires.